webhook_xblock/webhook_xblock.py

```diff
@@ -1,41 +1,90 @@
-        default=0, scope=Scope.user_state,
-        help="A simple counter, to show something happening",
-    )
+    checkpoint = Boolean(default=False, scope=Scope.user_state, help="Indicates if the student has already visited this course unit")
+    webhook_url = String(help="URL of the webhook to send the payload to", default=None, scope=Scope.content)
+    extra_info = Dict(help="Extra information to send in the payload", default={}, scope=Scope.content)
+    send_course_grade = Boolean(help="Indicates if the course grade should be sent in the payload", default=False, scope=Scope.content)
+    send_async = Boolean(help="Indicates if the payload should be sent in an async task", default=False, scope=Scope.content)
+    frequency = String(help="How often should the payload be sent", default="one-time", scope=Scope.content)
+    button_text = String(help="", default="Button inner HTML", scope=Scope.content)
+    component_text = String(help="Extra text to show in the component", default="", scope=Scope.content)
 
     def resource_string(self, path):
         """Handy helper for getting resources from our kit."""
         data = pkg_resources.resource_string(__name__, path)
         return data.decode("utf8")
 
-    # TO-DO: change this view to display your data your own way.
+    def check_if_running_from_studio(self):
+        """
+        Helper that returns True if the XBlock is currently
+        being called from studio mode, otherwise returns False.
+        """
+        is_studio = hasattr(self.xmodule_runtime, "is_author_mode")  # pylint: disable=no-member
+        if is_studio or self.xmodule_runtime.__class__.__name__ == "StudioEditModuleRuntime":
+            return True
+
+        return False
+
+    def make_request(self, data):
+        """
+        Helper for sending the request to the payload.
+        """
+        response = requests.post(self.webhook_url, data)
+
+        return response.status_code == 200
+
+    def mark_as_visited(self):
+        """
+        Helper called when the student visits the course unit with the
+        webhook component.
+        Returns True only when the frequency is "one-time",
+        and the unit has not been visited yet, otherwise
+        returns False.
+        """
+        if self.frequency == 'one-time' and not self.checkpoint:
+            self.checkpoint = True
+            return True
+
+        return False
+
     def student_view(self, context=None):
         """
         The primary view of the WebhookXblock, shown to students
         when viewing courses.
         """
         html = self.resource_string("static/html/webhook_xblock.html")
-        frag = Fragment(html.format(self=self))
+        frag = Fragment(html.format(
+            self=self, 
+            frequency=self.frequency, 
+            button_text=self.button_text,
+            component_text=self.component_text,
+        ))
         frag.add_css(self.resource_string("static/css/webhook_xblock.css"))
```

Comment:
do you have this file?

Reply:
@mariajgrimaldi yes it's the same from the cookie-cutter, so you can see it in the source code